clang/test/3C/typedefs.c

```diff
@@ -0,0 +1,49 @@
+// RUN: 3c -alltypes -addcr %s -- | FileCheck -match-full-lines -check-prefixes="CHECK_ALL","CHECK" %s
+// RUN: 3c -addcr %s -- | FileCheck -match-full-lines -check-prefixes="CHECK_NOALL","CHECK" %s
+// RUN: 3c --addcr --alltypes %s -- | %clang_cc1  -fcheckedc-extension -x c -
+// RUN: 3c --addcr %s -- | %clang_cc1  -fcheckedc-extension -x c -
+//
+// 3c -alltypes -output-postfix=checked %s
+// 3c -alltypes %S/typedefs.checked.c -- | diff %S/typedefs.checked.c -
+// rm %S/typedefs.checked.c
+
+typedef int* intptr;
+//CHECK: typedef _Ptr<int> intptr;
+
+typedef intptr* PP;
+//CHECK: typedef _Ptr<intptr> PP;
+
+typedef int* bad;
+//CHECK: typedef int* bad;
+
+typedef intptr* badP;
+//CHECK: typedef intptr* badP;
+
+typedef struct A { 
+    int* x;
+    int z;
+    PP p;
+} A;
+//CHECK: _Ptr<int> x;
+//CHECK: PP p;
+
+intptr bar(intptr x) { 
+	//CHECK: intptr bar(intptr x) _Checked {
+	return x;
+}
+
+int foo(void) { 
+    //CHECK: int foo(void) {
+    int x = 3;
+    intptr p = &x;
+    //CHECK: intptr p = &x;
+    PP pp = &p;
+    //CHECK: PP pp = &p;
+    A a = { &x, 3, pp };
+    //CHECK: A a = { &x, 3, pp };
+    bad b = (int*) 3;
+    //CHECK: bad b = (int*) 3;
+    badP b2 = (intptr*) 3;
```

Comment:
Does this compile with Checked C? In general, you should be having your tests run through the Checked C compiler after rewriting; it doesn't seem like that's happening?

Reply:
It does compile, and I believe the test does run it against clang on line 4. Let me know if I've set that up wrong.